src/try_extend.rs

```diff
@@ -0,0 +1,23 @@
+//! A trait for extending a collection with elements from an iterator, returning an error if the operation fails.
```

Comment:
I could not find a obvious file to add this trait so I created a new one. Suggestions are welcome.